### PR TITLE
Updated MultiMap utils javadoc and new unit tests

### PR DESCRIPTION
**Pull request by Skejven**
Basic test for `MultiMapConverter` util class usint `@TestWith` from `com.googlecode.zohhak`.

---

I hereby agree to the terms of the Knot.x Contributor License Agreement.

### CODE REVIEW

File: knotx-core/src/test/java/io/knotx/util/MultiMapConverterTest.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+      "mapKey1:A|mapKey2:B|mapKey3:C;{\"mapKey1\":[\"A\"],\"mapKey2\":[\"B\"],\"mapKey3\":[\"C\"]}",
+      "mapKey1:A,B|mapKey2:C,D|mapKey3:E,F;{\"mapKey1\":[\"A\",\"B\"],\"mapKey2\":[\"C\",\"D\"],\"mapKey3\":[\"E\",\"F\"]}",
+      "mapKey1:A,B,C|mapKey2:A,B,C|mapKey3:A,B,C;{\"mapKey1\":[\"A\",\"B\",\"C\"],\"mapKey2\":[\"A\",\"B\",\"C\"],\"mapKey3\":[\"A\",\"B\",\"C\"]}"
+  })
+  public void toJsonObject_whenMultiMapWithMultipleEntries_expectJsonObjectWithJsonArrays(
+      MultiMap multiMap, String expectedJson) throws Exception {
+    JsonObject jsonObject = MultiMapConverter.toJsonObject(multiMap);
+    assertThat(
+        jsonObject.toString(),
+        equalTo(expectedJson));
```
Comment (toniedzwiedz) [inline knotx-core/src/test/java/io/knotx/util/MultiMapConverterTest.java]: Using string comparison to make assertions regarding JSON objects can be cumbersome in the long run. Consider using  JSON-aware matchers. I've used [`hamcrest-json`](https://github.com/hertzsprung/hamcrest-json) previously and I was quite happy with the resutls.

File: knotx-core/src/main/java/io/knotx/util/DataObjectsUtil.java [hunk 2]
```diff
@@ -62,11 +62,11 @@ public static boolean equalsBody(io.vertx.core.buffer.Buffer self,
   }
 
   /**
-   * Method computing hashCode of the give Multimap.
-   * Multimap is treat as list of entires, key(String), value(List of Strings)
+   * Method computing hashCode of the give MultiMap.
+   * MultiMap is treat as list of entries, key(String), value(List of Strings)
```
Comment (toniedzwiedz) [inline knotx-core/src/main/java/io/knotx/util/DataObjectsUtil.java]: > MultiMap is treat as list of entries, key(String), value(List of Strings)

Something wrong with the grammar. The passive form of _treat_ is _treated_. How about:

> MultiMap is treated as a list of key-value pairs where each key is a String and the corresponding value is a List of Strings.

Comment (Skejven) [inline knotx-core/src/main/java/io/knotx/util/DataObjectsUtil.java]: Thanks, fixed.

File: knotx-core/src/test/java/io/knotx/util/MultiMapConverterTest.java (hunk 1 above)
Comment (Skejven) [inline knotx-core/src/test/java/io/knotx/util/MultiMapConverterTest.java]: Thank you, I didn't know about this lib. Those JSONs are pretty plain, however the syntax of `assertThat - sameJSONAs` is what fits here perfectly.

**Comment by tomaszmichalak** [comment]
Approved.

**Pull request merged**